src/fmu/dataio/datastructure/meta/content.py

```diff
@@ -181,200 +178,400 @@ class BoundingBox3D(BoundingBox2D):
 
 
 class Content(BaseModel):
-    content: enums.ContentEnum = Field(description="The contents of this data object")
+    """
+    The ``data`` block contains information about the data contained in this object.
+    This class is derived from for more specific content types that are discriminated
+    upon by the ``data.content`` field.
+    """
+
+    content: enums.ContentEnum
+    """The type of content these data represent."""
+
+    name: str = Field(examples=["VIKING GP. Top"])
+    """This is the identifying name of this data object. For surfaces, this is typically
+    the horizon name or similar. Shall be compliant with the stratigraphic column if
+    applicable.
+    """
 
     alias: Optional[List[str]] = Field(default=None)
+    """Other known-as names for ``data.name``. Typically names used within specific
+    software, e.g. RMS and others."""
+
+    tagname: Optional[str] = Field(
+        default=None,
+        examples=["ds_extract_geogrid", "ds_post_strucmod"],
+    )
+    """An identifier for this/these data object(s). Similar to the second part of the
+    generated filename in disk-oriented FMU data standard."""
```

Comment:
Add something like this: Please avoid using `tagname` as metadata in queries, since it's value is free-form. The intention with `tagname` is mostly backward compatibility with legacy scratch-file naming rules in FMU.